src/base/bittorrent/session.cpp

```diff
@@ -3044,13 +3049,17 @@ void Session::startUpTorrents()
         QByteArray data;
     } TorrentResumeData;
 
-    auto startupTorrent = [this, logger, resumeDataDir](const TorrentResumeData &params)
+    int resumedTorrentsCount = 0;
+    const auto startupTorrent = [this, logger, &resumeDataDir, &resumedTorrentsCount](const TorrentResumeData &params)
     {
         QString filePath = resumeDataDir.filePath(QString("%1.torrent").arg(params.hash));
         qDebug() << "Starting up torrent" << params.hash << "...";
         if (!addTorrent_impl(params.addTorrentData, params.magnetUri, TorrentInfo::loadFromFile(filePath), params.data))
             logger->addMessage(tr("Unable to resume torrent '%1'.", "e.g: Unable to resume torrent 'hash'.")
                                .arg(params.hash), Log::CRITICAL);
+
+        // process add torrent messages before message queue overflow
+        if (resumedTorrentsCount++ % 100 == 0) readAlerts();
```

Comment:
* Could you move the increment out? To after the `if` statement. It will become easier to read.
* Also prefer to use prefix increment
  IMHO, postfix increment adds a bit more burden when reading.